src/e2e/java/teammates/e2e/cases/lnp/BaseLNPTestCase.java

```diff
@@ -213,14 +277,18 @@ protected void runJmeter(boolean shouldCreateJmxFile) throws IOException {
         resultCollector.setFilename(resultsFile);
         testPlan.add(testPlan.getArray()[0], resultCollector);
 
-        // Run JMeter Test
+        // Run Jmeter Test
         jmeter.configure(testPlan);
         jmeter.run();
 
-        // TODO: As mentioned in the docs, good to fail the test if there is an Exception, or if the `success` value of
-        //  requests is `false`. An example of when this occurs is if the JMeter test properties are not set or if `email`
-        //  is used for logging in instead of `googleid`. Tests should fail if this assertion fails:
-        //      assertTrue(resultsErrorRate < this.getAcceptableErrorRate());
+        try {
+            ReportGenerator reportGenerator = new ReportGenerator(resultsFile, null);
+            reportGenerator.generate();
+        } catch (ConfigurationException | GenerationException e) {
+            log.warning(e.getMessage());
+        }
+
+        renameStatisticsFile();
```

Comment:
Will this cause race condition if multiple LNP is run concurrently?

Reply:
Are you referring to the renameStatisticsFile() method?

Reply:
As they all output to file call `Statistics.json`?

Reply:
I don't think multiple LNP tests should be run concurrently. If that happens, the results won't be accurate since the load on the system is higher than what would be shown in the results.

That being said, to be safe, we can still look into whether the default `statistics.json` file can use a dynamically generated filename instead (probably needs support from the Jmeter API or the `.properties` config file).

Reply:
> That being said, to be safe, we can still look into whether the default statistics.json file can use a dynamically generated filename instead (probably needs support from the Jmeter API or the .properties config file).

I think this is the correct why to tackle the problem regardless of whether we want to run it concurrently or not. Tests need to be **Independent**.

Reply:
Added timestamps to the temp files generated (and the statistics file) to avoid race conditions when running two tests concurrently, and additional functionality to make the tests remove old results (can be specified)

Reply:
Why we still have static filename here?

https://github.com/TEAMMATES/teammates/pull/10022/files#diff-6509e6ed4f2a3ea02c0d46c61b1ff10fR203

Reply:
Running the `ReportGenerator.generate()` makes the `ReportGenerator` class create an `statistics.json` file at the predefined path (LNP results folder in this case, predefined in [`reportgenerator.properties`](https://github.com/daekoon/teammates/blob/6ad72bde63269fef0fbd82faef2be902ff5537ae/src/e2e/resources/jmeter/bin/reportgenerator.properties#L196))

There seems to be no option to change the name of the statistics file generated from the `ReportGenerator`. (Checked `ReportGenerator` class, `reportgenerator.properties` file, `ResultCollector` class, `ExporterConfiguration` class and various API). Which leaves us with the next best option, to rename the generated statistics.json file.

I believe that the potential race condition caused by this static filename will be minimal. This file only gets generated once the LnP test is done during the report generation (doesn't take long), and then gets immediately renamed, so the duration for which this file is vulnuerable to race condition is minimal.